sources/api_client.go

```diff
@@ -11,48 +11,134 @@ import (
 
-	sourcesapi "github.com/lindgrenj6/sources-api-client-go"
+	"github.com/RedHatInsights/sources-api-go/model"
 	"github.com/redhatinsights/sources-superkey-worker/config"
 	l "github.com/redhatinsights/sources-superkey-worker/logger"
 )
 
 var xRhIdentity = `{"identity": {"account_number": "$ACCT$", "user": {"is_org_admin": true}}}`
 var conf = config.Get()
 
-// NewAPIClient - creates a sources api client with default header for account
-// returns: Sources API Client, error
-func NewAPIClient(identityHeader string) (*sourcesapi.APIClient, error) {
-	xrhid, err := getAccountNumber(identityHeader)
+func CheckAvailability(tenant string, sourceID string) error {
+	l.Log.Infof("Checking Availability for Source ID: %v", sourceID)
+
+	reqURL, _ := url.Parse(fmt.Sprintf(
+		"http://%v:%v/api/sources/v3.1/sources/%v", conf.SourcesHost, conf.SourcesPort, sourceID,
+	))
+
+	req := &http.Request{
+		Method: http.MethodPost,
+		URL:    reqURL,
+		Header: headers(tenant),
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil || resp.StatusCode != 202 {
+		return fmt.Errorf("failed to check availability for source %v: %v", sourceID, err)
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func CreateAuthentication(tenant string, auth *model.AuthenticationCreateRequest) error {
+	l.Log.Infof("Creating Authentication: %v", auth)
+
+	reqURL, _ := url.Parse(fmt.Sprintf(
+		"http://%v:%v/api/sources/v3.1/authentications", conf.SourcesHost, conf.SourcesPort,
+	))
+
+	body, err := json.Marshal(auth)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse x-rh-identity for SourcesApiClient: %v", identityHeader)
+		return err
 	}
 
-	// TODO: remove this once the PSK switchover is done.
-	if conf.SourcesPSK == "" {
-		return sourcesapi.NewAPIClient(&sourcesapi.Configuration{
-			DefaultHeader: map[string]string{"x-rh-identity": identityHeader},
-			Servers: []sourcesapi.ServerConfiguration{
-				{URL: fmt.Sprintf("%s://%s:%d/api/sources/v3.1", conf.SourcesScheme, conf.SourcesHost, conf.SourcesPort)},
-			},
-		}), nil
-	} else {
-		return sourcesapi.NewAPIClient(&sourcesapi.Configuration{
-			DefaultHeader: map[string]string{
-				"x-rh-sources-psk":            conf.SourcesPSK,
-				"x-rh-sources-account-number": xrhid,
-				"x-rh-identity":               identityHeader, // backwards compat -- needed for cost
-			},
-			Servers: []sourcesapi.ServerConfiguration{
-				{URL: fmt.Sprintf("%s://%s:%d/api/sources/v3.1", conf.SourcesScheme, conf.SourcesHost, conf.SourcesPort)},
-			},
-		}), nil
+	req := &http.Request{
+		Method: http.MethodPost,
+		URL:    reqURL,
+		Header: headers(tenant),
+		Body:   io.NopCloser(bytes.NewBuffer(body)),
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to create Authentication: %v", err)
+	} else if resp.StatusCode > 299 {
+		b, _ := io.ReadAll(resp.Body)
+		resp.Body.Close()
+		return fmt.Errorf("failed to create Authentication: %v", string(b))
+	}
+	defer resp.Body.Close()
```

Comment:
Nitpicking here, but shouldn't the deferral go inside the "else" block? Since otherwise the body is untouched.